File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int pairSum(ListNode* head) {
        int Max = -1;
        stack<int> twin;
        ListNode  *tmp = head;
        int i = 0;
        while(tmp && ++i) tmp = tmp->next;
        int mid = i/2;
        i = 0;
        for(int i = 0;head;i++)
        {
            if (i < mid)
                twin.push(head->val);
            else if (!twin.empty())
            {
                Max = max(Max,head->val + twin.top());
                twin.pop();
            }
             head = head->next;
        }
        return Max;
    }
};

File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {number}
 */
var pairSum = function(head) {
    var tmp = head;
    let Max = -1,i = 0,lent = 0;
    let stack = [];
    while(tmp != null && ++i)
        tmp = tmp.next;
    let mid = i/2;
    i = 0;
    while(head != null && ++i)
    {
        if (i <= mid)
            stack[lent++] = head.val;
        else
            Max = Math.max(Max, stack[--lent]+ head.val);
        head = head.next;
    }
    return Max;
    
};